src/models/TestStationDAO.ts

```
@@ -35,6 +35,34 @@ export class TestStationDAO {
     }
   }
 
+  /**
+   * Get test station for a given test station ID
+   * @param testStationPNumber
+   */
+  public async getTestStationByPNumber(testStationPNumber: string) {
+    const params = {
+      TableName: this.tableName,
+      FilterExpression: "#testStationPNumber = :testStationPNumber",
+      ExpressionAttributeNames: {
+        "#testStationPNumber": "testStationPNumber",
+      },
+      ExpressionAttributeValues: {
+        ":testStationPNumber": testStationPNumber,
+      },
+    };
+
+    const command = new ScanCommand(params);
+    const testStation = await TestStationDAO.dbClient.send(command);
+
+    if (!testStation || !testStation.Items || testStation.Count === 0) {
```

Would it be more appropriate to throw an error here, instead of returning an empty string? Looking at the returning function, it's being prompted to do data.Count but I don't think that you can do that on a string and might cause it to break and perhaps mask the validation issue

---

Changed it so that it returns null/undefined and handled it better so that we don't need to do a data.Count anymore.